encoding/masking-limiting-etc.md

```diff
@@ -83,22 +126,107 @@ Internally it uses a mask called a Camembert to generate a larger mask
 and limits it to the area affected by a line-darkening script.
 The main mask has no name and is simply dhhmask(mode=3).
 
-Example: Build a simple dehalo mask
+For more information about edgemasks, see [here](https://kageru.moe/blog/article/edgemasks).
+
+#### Example: Build a simple dehalo mask
+
+Suppose you want to remove these halos:
+
+![source](images/halos.png)
+
+![halos](images/src0.png)
+
+(Note that the images shown in your browser are likely resized poorly; 
+you can compare them at full size [here][comparison].)
+
+Fortunately, there is a well-established script that does just that: 
+[DeHalo_alpha][dehalo_alpha].
+
+However, we must be cautious in applying that filter,
+since, while removing halos reliably,
+it’s extremly destructive to the lineart as well.
+Therefore we must use a **dehalo mask**
+to protect the lineart and limit the filtering to halos.
+
+
+A dehalo mask aims to cover the halos
+but exclude the lines themselves,
+so that the lineart won’t be blurred or dimmed.
+In order to do that,
+we first need to generate an edgemask.
+In this example,
+we’ll use the built-in Sobel function.
+After generating the edge mask, we extract the luma plane:
+
+```py
+mask = core.std.Sobel(src, 0)
+luma = core.std.ShufflePlanes(mask, 0, colorfamily=vs.GRAY)
+
+```
+
+![mask luma](images/luma0.png)
+
+Next, we expand the mask twice, so that it covers the halos. 
+``vsutil.iterate`` is a function in [vsutil][vsutil iterate]
```

Comment:
If you want to refer to the iterate function in vsutil, the hyperlink needs to span `function in vsutil` as otherwise you'd expect to be referred to vsutil's main docs/page.